pyVHDLParser/Blocks/Reference/Context.py

```diff
@@ -27,139 +27,329 @@
+			parserState.NewBlock =      ReferenceEndBlock(parserState.LastBlock, parserState.TokenMarker, parserState.NewToken)
+			parserState.TokenMarker =   None
+			parserState.Pop()
+			return
+		elif isinstance(token, CharacterToken) and token == ",":
+			parserState.NewToken =      DelimiterToken(fromExistingToken=token)
+			parserState.NewBlock =      DelimiterBlock(parserState.LastBlock, parserState.TokenMarker, parserState.NewToken)
+			parserState.NextState =     cls.stateLibOrContextName
 			parserState.TokenMarker =   None
-			parserState.NextState =     cls.stateWhitespace2
 			return
 
-		raise BlockParserException("Expected whitespace after context name (identifier).", token)
+		raise BlockParserException("Expected either a comma or a semicolon separating or concluding a context reference (character).", token)
+
+
+@export
+class ReferenceEndBlock(Block):
+	pass
+
 
+
+#
+# Context declarations
+#
+@export
+class DeclarationStartBlock(Block):
 	@classmethod
-	def stateWhitespace2(cls, parserState: TokenToBlockParser):
+	def stateFromStartBlock(cls, parserState: TokenToBlockParser):
 		token = parserState.Token
-		if isinstance(token, WordToken) and (token <= "is"):
-			parserState.NewToken =      IsKeyword(fromExistingToken=token)
-			parserState.NewBlock =      cls(parserState.LastBlock, parserState.TokenMarker, endToken=parserState.NewToken)
-			parserState.NextState =     cls.stateDeclarativeRegion
-			return
-		elif isinstance(token, LinebreakToken):
-			if not (isinstance(parserState.LastBlock, CommentBlock) and isinstance(parserState.LastBlock.StartToken, MultiLineCommentToken)):
-				parserState.NewBlock =    cls(parserState.LastBlock, parserState.TokenMarker, endToken=token.PreviousToken, multiPart=True)
-				_ =                       LinebreakBlock(parserState.NewBlock, token)
-			else:
-				parserState.NewBlock =    LinebreakBlock(parserState.LastBlock, token)
-			parserState.TokenMarker =   None
+
+		if isinstance(token, ContextKeyword):
+			parserState.NextState =   cls.stateWhitespace
 			return
-		elif isinstance(token, CommentToken):
-			parserState.NewBlock =      cls(parserState.LastBlock, parserState.TokenMarker, endToken=token.PreviousToken, multiPart=True)
-			_ =                         CommentBlock(parserState.NewBlock, token)
-			parserState.TokenMarker =   None
+
+		# The 'context' keyword was already detected. If it is not there anymore, we found a bug.
+		assert False, "Expected keyword CONTEXT."
+
+	@classmethod
+	def stateContextKeyword(cls, parserState: TokenToBlockParser):
+		cls.stateWhitespace(parserState)
+
+	@classmethod
+	def stateWhitespace(cls, parserState: TokenToBlockParser):
+		if parserState.HandleNonCodeTokens(cls):
+			parserState.NextState =   cls.stateContextName
+			return
+
+		# This condition is also guaranteed. Otherwise `StartBlock.stateContextKeyword` would have thrown an error.
+		assert False, "Expected whitespace after keyword CONTEXT."

```

Comment:
I used failing assertions for unreachable code locations. If the interpreter gets there, there is either a bug somewhere, or the assessment that this location is unreachable is wrong.

If you do not want that kind of behaviour we can replace them with `BlockParserException`s for example.